contracts/governance/FeeSharingCollector/FeeSharingCollector.sol

```diff
@@ -937,6 +1041,41 @@ contract FeeSharingCollector is
         });
     }
 
+    /**
+     * @dev private function that responsible to calculate the user's token that has RBTC as underlying token (rbtc, wrbtc, iWrbtc)
+     *
+     * @param _token either RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT or wrbtc address or iwrbtc address
+     * @param _user address of the user.
+     * @param _maxCheckpoints maximum checkpoints.
+     *
+     * @return _tokenAmount token (rbtc, or wrbtc, or iwrbtc) amount
+     * @return _endToken end time of accumulated fee calculation for token (rbtc, or wrbtc, or iwrbtc )
+     */
+    function _getRBTCBalance(
+        address _token,
+        address _user,
+        uint32 _maxCheckpoints
+    ) private view returns (uint256 _tokenAmount, uint256 _endToken) {
+        IWrbtcERC20 wrbtcToken = protocol.wrbtcToken();
+
+        address loanPoolTokenWRBTC = _getAndValidateLoanPoolWRBTC(address(wrbtcToken));
```

Comment:
Currently in order to get the WRBTC Loan Pool we need to get the wrbtcToken from the protocol first.
And also it is being used redundantly because we need to check rbtc-based token related which can happen in both child and parent function.

I am thinking whether we need to store the loanTokenWrbtc into the FeeSharringCollectorStorage, will be much efficient, but we need to make sure to updattee everytime the loanTokenWrbtc address is changed which less likely to be happened.

Reply:
agreed  
created an optimization ticket